libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -586,11 +636,37 @@ void AP_Baro::update(void)
         }
     }
 
+    // consider a sensor as healthy if it has had an update in the
+    // last 0.5 seconds
+    uint32_t now = AP_HAL::millis();
+    for (uint8_t i=0; i<_num_sensors; i++) {
+        sensors[i].healthy = (now - sensors[i].last_update_ms < 500) && !is_zero(sensors[i].pressure);
+    }
+
+    // find current GPS altitude
+    float gps_alt = 0.0f;
+    bool process_gps_alt = false;
+    if (!is_zero(_gps_adj_step) && _gps_calibrated) {
+        float gps_vacc;
+        if (AP_GPS::gps().vertical_accuracy(gps_vacc)) {
+            const uint32_t last_fix = AP_GPS::gps().last_fix_time_ms();
+
+            if (gps_vacc < BARO_MAX_GPS_ACCURACY && (now - last_fix) < BARO_MAX_GPS_DELAY) {
```

Comment:
Again just ask the GPS if it's healthy, rather then applying your own time limit here. It will catch more.

Reply:
See above.